File: Modules/TaskFlow/Task/TaskForm.cs
namespace TaskFlowSerenity.TaskFlow.Forms;

[FormScript("TaskFlow.Task")]
[BasedOnRow(typeof(TaskRow), CheckNames = true)]
public class TaskForm
{
    public Guid ProjectId { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public Guid TaskStateId { get; set; }
    public DateTime DueDate { get; set; }

    [DisplayName("Tags")]
    public List<Guid> TagIds { get; set; }
}

File: Modules/ServerTypes/TaskFlow/TaskForm.ts
import { EnumEditor, initFormType, IntegerEditor, LookupEditor, PrefixedContext, ServiceLookupEditor, StringEditor } from "@serenity-is/corelib";
import { PersianDateEditor } from "../../Common/PersianDateEditor";
import { TaskPriority } from "./TaskPriority";

export interface TaskForm {
    ProjectId: ServiceLookupEditor;
    Title: StringEditor;
    Description: StringEditor;
    TaskStateId: LookupEditor;
    Priority: EnumEditor;
    DueDate: PersianDateEditor;
    Duration: IntegerEditor;
    TagIds: ServiceLookupEditor;
}

export class TaskForm extends PrefixedContext {
    static readonly formKey = 'TaskFlow.Task';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!TaskForm.init) {
            TaskForm.init = true;

            initFormType(TaskForm, [
                'ProjectId', ServiceLookupEditor,
                'Title', StringEditor,
                'Description', StringEditor,
                'TaskStateId', LookupEditor,
                'Priority', EnumEditor,
                'DueDate', PersianDateEditor,
                'Duration', IntegerEditor,
                'TagIds', ServiceLookupEditor
            ]);
        }
    }
}

[TaskPriority]; // referenced types